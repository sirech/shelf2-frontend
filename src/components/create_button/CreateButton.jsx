// @flow

import React from 'react'
import { Button } from 'reactstrap'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

import styles from './styles.module.css'

const CreateButton = ({ onClick }: { onClick: (void) => void }) => (
  <span className={styles.createButton}>
    <Button color="success" onClick={onClick}>
      <FontAwesomeIcon icon={faPlus} />
      Add Book
    </Button>
  </span>
)

export default CreateButton
