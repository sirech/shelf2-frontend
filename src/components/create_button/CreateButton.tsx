// @flow

import React from 'react'
import { Button } from 'reactstrap'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

import styles from './styles.module.css'

const CreateButton = ({ onClick }: { onClick: () => void }) => (
  <span className={styles.createButton}>
    <Button color="success" onClick={onClick}>
      <FontAwesomeIcon icon={faPlus} />
      Add Book
    </Button>
  </span>
)

export default CreateButton
