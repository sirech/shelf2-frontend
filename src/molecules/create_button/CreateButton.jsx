// @flow

import React from 'react'
import { Button } from 'reactstrap'

import { FaPlus } from 'react-icons/fa'

import './styles.css'

const CreateButton = ({ onClick }: { onClick: void => void }) => (
  <span className="create-button">
    <Button color="success" onClick={onClick}>
      <FaPlus />
      Add Book
    </Button>
  </span>
)

export default CreateButton
