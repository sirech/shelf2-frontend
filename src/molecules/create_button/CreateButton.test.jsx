import React from 'react'
import renderer from 'react-test-renderer'
import CreateButton from './CreateButton'

describe('components', () => {
  describe('CreateButton', () => {
    it('renders correctly', () => {
      const component = renderer.create(<CreateButton />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
