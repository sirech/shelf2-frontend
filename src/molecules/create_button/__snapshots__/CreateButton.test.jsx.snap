// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components CreateButton renders correctly 1`] = `
<span
  className="create-button"
>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type={undefined}
  >
    <svg
      attr={
        Object {
          "viewBox": "0 0 448 512",
        }
      }
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      />
    </svg>
    Add Book
  </button>
</span>
`;
